### PR TITLE
[ENH]: Bump dependencies

### PR DESCRIPTION
**Pull request by fraimondo**
### Are you requiring a new dataset or marker?

- [X] I understand this is not a marker or dataset request

### Which feature do you want to include?

New packages have been published for:

`numpy`: 1.26.0
`datalad`: 0.19.3
`pandas`: 2.1.1
`nibabel`: 5.10
`nilearn`: 0.10.1
`sqlalchemy`: 2.0.21
`h5py`: 3.9

Also testing/docs packages versions should be updated.

### How do you imagine this integrated in junifer?

Update `pyproject.toml`

### Do you have a sample code that implements this outside of junifer?

_No response_

### Anything else to say?

_No response_

### CODE REVIEW

File: docs/changes/newsfragments/253.change
[the remark this answers is not in the document]
Comment (fraimondo) [reply docs/changes/newsfragments/253.change:1]: done

[the remark this answers is not in the document]
Comment (synchon) [reply docs/changes/newsfragments/253.change:1]: The file still shows up as `.change`.

File: conda-env.yml [hunk 1]
```diff
@@ -3,23 +3,23 @@ channels:
...
-  - sphinx-copybutton=0.5.*
-  - towncrier=22.12.*
-  - sphinxcontrib-mermaid=0.8.*
+  - h5py>=3.8,<=3.9
+  - seaborn>=0.11,<=0.13
+  - sphinx>=5.3,<7.3
+  - sphinx-gallery>=0.11.0,<0.15.0
+  - furo>=2022.9.29,<2023.10.0
+  - numpydoc>=1.5.0,<1.6
+  - phinx-copybutton>=0.5.1,<0.5.3
```
Comment (synchon) [inline conda-env.yml:20]: `sphinx` and not `phinx`?

Comment (fraimondo) [reply conda-env.yml]: bad copy paste

**Pull request merged**